meraki/__init__.py

```diff
@@ -167,20 +170,21 @@ def __init__(self,
             use_iterator_for_get_pages=use_iterator_for_get_pages,
         )
 
+        self._enable_kwargs_validation = enable_kwarg_validation
         # API endpoints by section
-        self.administered = Administered(self._session)
-        self.organizations = Organizations(self._session)
-        self.networks = Networks(self._session)
-        self.devices = Devices(self._session)
-        self.appliance = Appliance(self._session)
-        self.camera = Camera(self._session)
-        self.cellularGateway = CellularGateway(self._session)
-        self.insight = Insight(self._session)
-        self.licensing = Licensing(self._session)
-        self.sensor = Sensor(self._session)
-        self.sm = Sm(self._session)
-        self.switch = Switch(self._session)
-        self.wireless = Wireless(self._session)
+        self.administered = Administered(self._session, self._enable_kwargs_validation )
+        self.organizations = Organizations(self._session, self._enable_kwargs_validation )
+        self.networks = Networks(self._session, self._enable_kwargs_validation )
+        self.devices = Devices(self._session, self._enable_kwargs_validation )
+        self.appliance = Appliance(self._session, self._enable_kwargs_validation )
+        self.camera = Camera(self._session, self._enable_kwargs_validation )
+        self.cellularGateway = CellularGateway(self._session, self._enable_kwargs_validation )
+        self.insight = Insight(self._session, self._enable_kwargs_validation )
+        self.licensing = Licensing(self._session, self._enable_kwargs_validation )
+        self.sensor = Sensor(self._session, self._enable_kwargs_validation )
+        self.sm = Sm(self._session, self._enable_kwargs_validation )
+        self.switch = Switch(self._session, self._enable_kwargs_validation )
+        self.wireless = Wireless(self._session, self._enable_kwargs_validation )
 
         # Batch definitions
-        self.batch = Batch()
+        self.batch = Batch(self._enable_kwargs_validation )
```

Comment:
This should not be needed. None of the other config kwargs receive this treatment and we should not start treating kwargs this way.